Fix build for gimp-with-plugins against exiv2 0.27.1

gloaming opened the pull request:
###### Motivation for this change

Fixes #64586

###### Things done

Path size `1803392632` vs `1798388408` when building against `exiv2 0.26.2018.12.30`

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

gloaming committed:
gimpPlugins.gimplensfun: 0.2.4 -> unstable-2018-10-21

Fixes broken build with exiv2-0.27.1 by using gexiv2 instead

worldofpeace commented on pkgs/applications/graphics/ufraw/default.nix:
```diff
@@ -13,6 +13,17 @@ stdenv.mkDerivation rec {
     sha256 = "0pm216pg0vr44gwz9vcvq3fsf8r5iayljhf5nis2mnw7wn6d5azp";
   };
 
+  patches = [
```
See review at #64353.

Those patches could be improved.

gloaming replied:
Roger that.
@neilmayhew could you test the suggested changes and update the patches at sourceforge.net?

neilmayhew replied:
@gloaming Done. The URLs are:

* https://sourceforge.net/p/ufraw/bugs/_discuss/thread/c0761df338/a2d4/attachment/0001-Change-use-of-header-files.patch
* https://sourceforge.net/p/ufraw/bugs/_discuss/thread/c0761df338/a2d4/attachment/0002-Use-symbolic-error-code.patch

Not as tidy as before, but I don't think SourceForge provides a way to modify attachments in the issue description. I had to do it by attaching the new files to a comment.

gloaming replied:
Thanks! Sorry I didn't notice your PR before opening this one.

neilmayhew replied:
> Sorry I didn't notice your PR before opening this one.

No worries. It might have helped if I'd created an issue as well as a PR. I'll make sure to do that in future.

gloaming commented:
Hold up though - I forgot to actually test `ufraw` properly by opening a photo before. I downloaded a couple to check and it seems the release version is broken!

I dug around a bit more and found that someone is maintaining a fork on GitHub: https://github.com/sergiomb2/ufraw

It includes several distro patches, including the `exiv2` fix: https://github.com/sergiomb2/ufraw/commit/a13c0f7385ff873bd7cf8fc8aae8657c75fb20c8
I'm going to go ahead and update to HEAD there, since at least I can actually open one test photo with it, although not the other.

worldofpeace commented:
That's sounds fine, I've checked and fedora also uses that fork.

gloaming committed:
ufraw: 0.22 -> unstable-2019-06-12

The original ufraw repo is no longer maintained, so change
to a new upstream that's collecting patches, including one
we need to fix the build against more recent versions of exiv2.

gloaming commented:
@worldofpeace It appears that GIMP has not supported the `ufraw` plugin for some time. Should we remove all mention of `ufraw.gimpPlugin`?

worldofpeace commented:
@gloaming I believe that should be fine. If you do this please note you need to add a `throw` to `aliases.nix` on `gimpPlugins.ufraw` to explain the backwards incompatible change.

worldofpeace commented:
ping @gloaming 

(sorry this is dragging on so long)

jtojnar commented:
Ufraw was working recently to some capacity at least: https://github.com/NixOS/nixpkgs/pull/37915 What exactly does “GIMP not supporting plug-in” mean?

gloaming commented:
@jtojnar `gimpWithPlugins` doesn't have any trace of `ufraw` in its menus. I tried to open a RAW that opens with `ufraw` inside GIMP, and it gave me the same message it gives without the `ufraw` plugin:

![gimp-raw-loaders](https://user-images.githubusercontent.com/10156748/62195215-68a89380-b373-11e9-9df8-88f9a0bbec03.png)

The message itself implies that `ufraw` is no longer supported.

gloaming commented:
> @gloaming I believe that should be fine. If you do this please note you need to add a `throw` to `aliases.nix` on `gimpPlugins.ufraw` to explain the backwards incompatible change.

@worldofpeace Hmm... Would it be better to put the `throw` in `pkgs/applications/graphics/gimp/plugins/default.nix` in case some future traveller looks for it there?

gloaming commented:
Actually, I think it's better to leave it for now. Rebased and checked it still works. @worldofpeace feel free to merge

jtojnar commented:
Okay, I can confirm that opening RAW files through the file dialogue no longer works but the send to GIMP button in ufraw seems to still require the plug-in.

gloaming commented:
Ah, right. Happens the same for me. I've left it at just the updates then.

jtojnar commented on pkgs/applications/graphics/ufraw/default.nix:
```diff
@@ -1,38 +1,72 @@
 
-  postInstall = stdenv.lib.optionalString withGimpPlugin ''
+  postInstall = ''
+    rm -f $out/bin/dcraw
```
Why are you deleting this file? Could you add a simple comment here and a more detailed description into the commit message?

gloaming replied:
The problem is the `--enable-extras` option. From the README:
```
--enable-extras: build the extra binaries - dcraw, nikon-curve.
```
```
For Packagers
=============
You are free to package UFRaw in any way, as long as it complies with the
GNU GPL. The followings are just recommendations.

Do not package the executables generated by by --enable-extras. These extras
are there for testing the code during development. They are of no interest
to end user. Specifically, if you want to package dcraw, you should use
Dave's original code and not UFRaw's modified code.
```
However, the existing derivation was already compiling with this option; I took this to mean that somebody wanted the `nikon-curve` binary for some reason, so I left it in to be conservative.

Checking the commit history, I see it was you that added the option :) so if you're not aware of any use case, I think it would be better to remove `--enable-extras` and see if anyone complains.

jtojnar replied:
I merely copied the flags from https://github.com/NixOS/nixpkgs/commit/63a234c4a720f3457801043276e1884e451d3b85#diff-9120f8657759d45707b61eaf16c44e4bL188-L196. The [original packager](https://github.com/NixOS/nixpkgs/commit/b241732d8cd5c6b22bfd8f2a256d33790929f5c5) must have added it for the sake of completeness, but since the binaries would not be available when used as a GIMP plug-in anyway, there should be no reason to include extras for a plug-in. Could you drop it and mention the reasons in the commit message?

gloaming replied:
Done.

jtojnar commented on pkgs/applications/graphics/ufraw/default.nix:
```diff
@@ -1,38 +1,72 @@
-    # XXX: These guys appear to mutate uploaded tarballs!
-    url = "mirror://sourceforge/ufraw/${name}.tar.gz";
-    sha256 = "0pm216pg0vr44gwz9vcvq3fsf8r5iayljhf5nis2mnw7wn6d5azp";
+  src = fetchFromGitHub {
```
Please add a comment with explanation and a more detailed description of the reason for switching to the fork, in the commit message.

ajs124 commented on pkgs/applications/graphics/ufraw/default.nix:
```diff
@@ -1,38 +1,72 @@
+    zlib
   ] ++ stdenv.lib.optional withGimpPlugin gimp;
 
+  preConfigure = "./autogen.sh";
```
Shouldn't `autoreconfHook` be able to do this?

jtojnar replied:
Possibly. There does not seem to be anything special in the file: https://github.com/sergiomb2/ufraw/blob/c65b4237dcb430fb274e4778afaf5df9a18e04e6/autogen.sh#L10-L13, so I would expect `autoreconf` to deal with it.

gloaming replied:
Thanks, didn't know about that one.

gloaming committed:
ufraw: Remove --enable-extras

Don't build the 'dcraw' and 'nikon-curve' binaries, as per ufraw's README:

"Do not package the executables generated by by --enable-extras. These extras
are there for testing the code during development. They are of no interest
to end user. Specifically, if you want to package dcraw, you should use
Dave's original code and not UFRaw's modified code."

gloaming commented:
Ping @worldofpeace

worldofpeace commented:
Thanks, sorry about the delay on this.
